src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -115,27 +115,36 @@ public static void main(String[] args) throws Exception {
 
-        try (InputStream rawBase64 = factory.readDictionary();
-             ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(Base64
-                     .getDecoder()
-                     .decode(rawBase64.readAllBytes()));
-             ZipInputStream zipInputStream = new ZipInputStream(byteArrayInputStream)) {
-            ZipEntry zipEntry = null;
-            try {
-                while ((zipEntry = zipInputStream.getNextEntry()) != null) {
-                    Path filePath = Path.of(zipEntry.getName());
-                    if (!filePath.toString().endsWith(".xml")) {
-                        throw new IncorrectFixFileNameException("Incorrect FIX dictionary file name: " + filePath.getFileName());
+        if (settings.isZipDictionaries()) {
+            try (InputStream rawBase64 = factory.readDictionary();
+                 ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(Base64
+                         .getDecoder()
+                         .decode(rawBase64.readAllBytes()));
+                 ZipInputStream zipInputStream = new ZipInputStream(byteArrayInputStream)) {
+                ZipEntry zipEntry = null;
+                try {
+                    while ((zipEntry = zipInputStream.getNextEntry()) != null) {
+                        Path filePath = Path.of(zipEntry.getName());
+                        if (!filePath.toString().endsWith(".xml")) {
+                            throw new IncorrectFixFileNameException("Incorrect FIX dictionary file name: " + filePath.getFileName());
+                        }
+                        writeDictionary(temporaryDirectory, zipInputStream, filePath);
                     }
-                    Path pathToDictionary = Files.createFile(getPathToDictionary(temporaryDirectory, filePath));
-                    Files.write(pathToDictionary, zipInputStream.readAllBytes());
-                    new DataDictionary(pathToDictionary.toString()); //check that xml file contains the correct values
+                } catch (IncorrectFixFileNameException e) {
+                    throw new Exception("Failed to unzip dictionaries along the path: " + zipEntry.getName(), e);
+                }
+            } catch (Exception e) {
+                throw new Exception("Failed to create DataDictionary", e);
+            }
+        } else {
+            try {
+                for (String dictionaryAlias : factory.getDictionaryAliases()) {
+                    InputStream dictionary = factory.loadDictionary(dictionaryAlias);
+                    writeDictionary(temporaryDirectory, dictionary, Path.of(dictionaryAlias));
                 }
-            } catch (IncorrectFixFileNameException | ConfigError e) {
-                throw new Exception("Failed to unzip dictionaries along the path: " + zipEntry.getName(), e);
+            } catch (Exception e) {
+                throw new Exception("Failed to create DataDictionary", e);
             }
```

Comment:
I'd suggest wrapping cycle-body into try-catch instead. This way we'll have info about which dictionary we've failed to load